staging/src/k8s.io/kubectl/pkg/cmd/create/create_service.go

```diff
@@ -45,6 +54,156 @@ func NewCmdCreateService(f cmdutil.Factory, ioStreams genericclioptions.IOStream
 	return cmd
 }
 
+// ServiceOpts holds the options for 'create service' sub command
+type ServiceOpts struct {
+	PrintFlags *genericclioptions.PrintFlags
+	PrintObj   func(obj runtime.Object) error
+
+	Name         string
+	TCP          []string
+	Type         v1.ServiceType
+	ClusterIP    string
+	NodePort     int
+	ExternalName string
+
+	FieldManager     string
+	CreateAnnotation bool
+	Namespace        string
+
+	Client         clientCoreV1.CoreV1Interface
+	DryRunStrategy cmdutil.DryRunStrategy
+	DryRunVerifier *resource.DryRunVerifier
+	genericclioptions.IOStreams
+}
+
+// Complete completes all the required options
+func (o *ServiceOpts) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
+	name, err := NameFromCommandArgs(cmd, args)
+	if err != nil {
+		return err
+	}
+	o.Name = name
+
+	cs, err := f.KubernetesClientSet()
```

Comment:
You should create the client you care about, so `corev1client.NewForConfig(clientConfig)` see other files for examples.

Reply:
This is still unresolved.

Reply:
Hi! Please point me to exactly one file that has the code the way you expect it.
Because I seem to only find examples that looks similar to the code I already have:
For example staging/src/k8s.io/kubectl/pkg/cmd/create/create_role.go:275
But I think your are talking about something like:
staging/src/k8s.io/kubectl/pkg/cmd/create/create_job.go:128
correct?

Reply:
https://github.com/kubernetes/kubernetes/blob/f15e9f6fce10a138b0b11641cb8b18c327131ed3/staging/src/k8s.io/kubectl/pkg/cmd/create/create_job.go#L128 is one such example.

Reply:
I think I got it right now.